### PR TITLE
added loadContract overload for finer control over contract object

### PR DESCRIPTION
**Pull request by afremuth-figure**
## Description
The ability to load and execute requested (fragment) contracts was required.  This PR contains an overload to `loadContract` with this increased flexibility.

## Files most critical to review
`p8e-sdk/src/main/kotlin/io/p8e/ContractManager.kt`

### CODE REVIEW

File: p8e-sdk/src/main/kotlin/io/p8e/ContractManager.kt [hunk 1]
```diff
@@ -328,17 +328,45 @@ class ContractManager(
...
-            contractClassExecutor(clazz),
+            contractClazz,
+            contractClassExecutor(contractClazz),
+            isFragment
+        )
+    }
+
+    /**
+     * Returns a new Contract object that can be configured and executed. This Contract represents a contract
+     * based on a previously executed Contract that did not complete successfully. The envelope will contain
```
Comment (piercetrey-figure) [inline p8e-sdk/src/main/kotlin/io/p8e/ContractManager.kt:357]: Isn't this comment backwards? Isn't this for loading a previously executed contract (i.e. first step executed successfully, loading the contract for the second step execution)

Comment (afremuth-figure) [reply p8e-sdk/src/main/kotlin/io/p8e/ContractManager.kt:357]: ahh yep, I'll make the update

File: p8e-sdk/src/main/kotlin/io/p8e/ContractManager.kt [hunk 2]
```diff
@@ -328,17 +328,45 @@ class ContractManager(
 //            }
 //    }
 
+
+    /**
+     * Returns an existing Contract object that can be configured and executed.
+     *
+     * @param contractClazz subclass of P8eContract that represents the original contract
+     * @param executionUuid the execution id of the original contract
+     * @param isFragment flag identifying if the original contract execution was requested (true) or initiated (false)
```
Comment (piercetrey-figure) [inline p8e-sdk/src/main/kotlin/io/p8e/ContractManager.kt:337]: This isFragment param comment confuses me, but I'm not entirely sure how to make it better 😆

Comment (piercetrey-figure) [reply p8e-sdk/src/main/kotlin/io/p8e/ContractManager.kt:337]: maybe something like 'flag identifying whether this contract is to be a response to a previous execution request (true) or a fresh execution (false)'

**Review by piercetrey-figure — APPROVED** [review]
code-wise looks good, so once the comments are addressed 👍

**Pull request merged**